NDTensors/ext/examples/NDTensorCUDA.jl

```diff
@@ -1,10 +1,10 @@
-using CUDA
 using NDTensors
-
-using ITensors
-using Test
-
-using Zygote
+using CUDA: cu, CUDA, CuVector
```

Comment:
```suggestion
using CUDA: CUDA, CuVector, @allowscalar, cu
```
In the examples we it's better to use `@allowscalar` from `CUDA` instead of from `GPUArraysCore`, that's what most users would be doing. Internally it makes sense to try to use `GPUArraysCore` since it is a lighter weight dependency, but for end users they will be using packages like `CUDA`, `Metal`, etc. that ship with `@allowscalar`, so there isn't a need to load `GPUArraysCore` separately.

Reply:
Yeah that makes sense, we also are sure that CUDA and Metal are around in their examples which we aren't sure of in the generic code.  I made these changes and also changed the next line to be variables first then functions, thanks for that reminder too.